### DISCUSSION TITLE
Does wxt work with TypeScript 4?

### DISCUSSION THREAD

**Discussion by ferenc-a** [wxt-dev/wxt#1212] [Q&A]
Running `wxt build` works fine in a TypeScript 4 monorepo, but running `tsc --noEmit` (which is present in the init project scripts by the name of `compile`) throws an error. How can `build` produce a result even though `compile` fails?
My issue is probably caused by TypeScript not having the bundler option for module resolution, which is set in wxt's generated tsconfig. Some comments I found on the repo also reference TypeScript 5 feature (like extending multiple configs: https://github.com/wxt-dev/wxt/discussions/572#discussioncomment-10595910, https://github.com/wxt-dev/wxt/issues/868#issuecomment-2363900402).

Does this package supprt TypeScript, is there anything I can configure for it to work? If not, this should be documented somewhere. If not on the main readme, at least somewhere like here: https://wxt.dev/guide/essentials/config/typescript#typescript-configuration.

**Comment by aklinker1** [comment]
> How can build produce a result even though compile fails?

Because WXT strips out all types during the build process without checking them. This is very standard for build frameworks, because typescript is really slow.

> My issue is probably caused by TypeScript not having the bundler option for module resolution, which is set in wxt's generated tsconfig

You should be able to override it in your root tsconfig, the one that extends the one generated by WXT. If that doesn't work, you could use the `prepare:types` hook to modify the tsconfig text content before it's written to the disk.

**Comment by ferenc-a** [comment]
I had other problems with the `moduleResolution` workaround, and I have no idea what `prepare:types` is, nor could I find it by Googling.
I managed to update to TS 5 in the end, my issue is done, and I'm not sure what would have fixed it with TS 4.

**Reply by aklinker1** [reply]
For future reference: https://wxt.dev/api/reference/wxt/interfaces/WxtHooks.html#prepare-types
